virttest/qemu_vm.py

```diff
@@ -1106,8 +1106,41 @@ def add_qxl(qxl_nr, base_addr=29):
                 qxl_str += " -device qxl,id=video%d,addr=0x%x" % (index, addr)
             return qxl_str
 
-        def add_vga(vga):
-            return " -vga %s" % vga
+        def add_vga(devices, vga):
+            """Add primary vga device."""
+            default_machine_re = re.compile(".*")
+            isapc_machine_re = re.compile("isapc")
+            pci_bus = {'aobject': 'pci.0'}
+            isa_bus = None  # so far there is no ISA bus
+            vga_dev_map = {
+                # name: [(machine regex, driver, parent bus), ...]
+                # the default variant must be the last one
```

Comment:
I'd suggest add a comment here, things like:
`# adding isapc support, the default variant must be the last one, so if there is no match, use default values`